tests/system.rs

```diff
@@ -485,3 +503,142 @@ fn string_eq_ignore_newline_repr_test() {
+//   "## `NAME`".
+// - Configuration values in Configurations.md must be in the form of
+//   "#### `VALUE`".
+fn get_code_blocks() -> Vec<CodeBlock> {
+    let filename = "Configurations.md";
+    let filebuf = BufReader::new(
+        fs::File::open(filename).expect(&format!("Couldn't read file {}", filename)),
+    );
+
+    let config_name_regex =
+        regex::Regex::new(r"^## `([^`]+)`").expect("Failed creating configuration pattern");
+    let config_value_regex = regex::Regex::new(r#"^#### `"?([^`"]+)"?`"#)
+        .expect("Failed creating configuration value pattern");
+
+    let mut code_blocks = Vec::new();
+    let mut extracting_block = false;
+    let mut code_block = Vec::new();
+    let mut code_block_start_line = 0;
+    let mut config_name: Option<String> = None;
+    let mut config_value: Option<String> = None;
+
+    for (i, line) in filebuf.lines().map(|l| l.unwrap()).enumerate() {
+        if line.starts_with("```") && extracting_block {
+            // We've reached the end of the code block.
+            extracting_block = false;
+
+            // Get a trailing newline in the code block.
+            code_block.push(String::new());
+
+            // Store the code block information.
+            code_blocks.push(CodeBlock {
+                config: ConfigPair {
+                    name: config_name.as_ref().unwrap().clone(),
+                    value: config_value.unwrap(),
+                },
+                code_block: code_block.join("\n"),
+                code_block_start_line,
+            });
+
+            config_value = None;
+            code_block.clear();
+        } else if line.starts_with("```rust") {
+            // We've reached the beginning of a Rust code block.
+            extracting_block = true;
+
+            // +1 to translate to one-based indexing
+            // +1 to get to first line of code (line after "```")
+            code_block_start_line = (i + 2) as u32;
+
+            // Assert that we have a configuration name and value.
+            assert!(
+                config_name.is_some(),
+                "configuration name not found for block beginning at line {}",
+                code_block_start_line
+            );
+            assert!(
+                config_value.is_some(),
+                "configuration value not found for block beginning at line {}",
+                code_block_start_line
+            );
+        } else if extracting_block {
+            code_block.push(line);
+        } else if let Some(c) = config_name_regex.captures(&line) {
+            config_name = Some(String::from(&c[1]));
+        } else if let Some(c) = config_value_regex.captures(&line) {
+            config_value = Some(String::from(&c[1]));
+        }
+    }
+
+    code_blocks
+}
+
+// Format the code blocks extracted in `get_code_blocks` and check for
+// failures.
+fn check_blocks_idempotency(blocks: &Vec<CodeBlock>) {
+    let mut failures = 0;
+
+    for block in blocks {
```

Comment:
Optional: this loop might work better as a `fold` if you can factor out some of the code into little functions, not sure though.